addons/pinniped/config-controller/hack/run.sh

```diff
@@ -5,22 +5,141 @@
 ./hack/generate.sh
-kubectl apply -f "${RBAC_FILE}"
 
-docker build -t "$image" .
-docker push "$image"
-ytt --data-value "image=$image" -f "${DEPLOYMENT_FILE}" | kbld -f - | kapp deploy -a pinniped-config-controller-manager -f - -y
-kubectl logs -n pinniped deploy/pinniped-config-controller-manager -f
+# the namespace must exist before the package can be deployed
+echo -e "${YELLOW}creating namespace ${APP_NAME}...${DEFAULT}"
+ytt --data-value "tanzuAuth.namespace=${APP_NAME}" --file "${TANZU_AUTH_CONFIG_DIR}/namespace.yaml" | kubectl apply -f -
+
+echo -e "${YELLOW}rebuilding image lock file with ytt and kbld...${DEFAULT}"
+ytt --data-value "image=${CONTROLLER_IMAGE}" --file "${TANZU_AUTH_DEFAULT_VALUES_FILE}" --file "${TANZU_AUTH_CONFIG_DIR}" \
+  | kbld --file - --imgpkg-lock-output "${TANZU_AUTH_BUNDLE_DIR}/.imgpkg/images.yml"
+
+# push the tanzu-auth-controller-manager-package to the registry
+echo -e "${YELLOW}pushing ${PACKAGE_IMAGE} to registry with imgpkg...${DEFAULT}"
+imgpkg push --bundle "${PACKAGE_IMAGE}" --file "${TANZU_AUTH_BUNDLE_DIR}"
+
+# inject the package name into the package file & deploy
+echo -e "${YELLOW}creating package ${TANZU_AUTH_PACKAGE_FILE} with image ${PACKAGE_IMAGE}...${DEFAULT}"
+yq e ".spec.template.spec.fetch[0].imgpkgBundle.image = \"${PACKAGE_IMAGE}\"" "${TANZU_AUTH_PACKAGE_FILE}" \
+  | kubectl apply -f -
+kubectl apply -f "${TANZU_AUTH_DIR}/metadata.yaml"
+
+
+# Deploy the package on the cluster
+echo -e "${YELLOW}creating package install...${DEFAULT}"
+PACKAGE_SA_NAME="${CONTROLLER_MANAGER_PACKAGE_NAME}-sa"
+PACKAGE_NAMESPACE="tkg-system"
+PACKAGE_INSTALL="$(cat <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ${PACKAGE_SA_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: ${PACKAGE_SA_NAME}-cluster-admin
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: ${PACKAGE_SA_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${CONTROLLER_MANAGER_PACKAGE_NAME}-config
+  namespace: ${PACKAGE_NAMESPACE}
+stringData:
+  values.yaml: |
+    tanzuAuth:
+      namespace: ${CONTROLLER_NAMESPACE_NAME}
+    image: ${CONTROLLER_IMAGE}
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  name: ${APP_NAME}
+  namespace: ${PACKAGE_NAMESPACE}
+spec:
+  packageRef:
+    refName: ${APP_NAME}.tanzu.vmware.com
+    versionSelection:
+      constraints: "1.6.0"
+      prereleases: {}
+  syncPeriod: 30s
+  serviceAccountName: ${PACKAGE_SA_NAME}
+  values:
+  - secretRef:
+      name: ${CONTROLLER_MANAGER_PACKAGE_NAME}-config
+EOF)"
+
+echo -e "${YELLOW}package install contents...${DEFAULT}"
+echo "${PACKAGE_INSTALL}"
+
+echo -e "${YELLOW}applying package install contents...${DEFAULT}"
+echo "${PACKAGE_INSTALL}" | kubectl apply -f -
```

Comment:
Per our zoom convo, this approach is more like what kapp-controller will do on the cluster, rather than running the ytt,kbld and piping the yaml outputs straight to local kapp.  [link to the original spike code](https://github.com/ankeesler/tanzu-framework/pull/4/files#diff-d547f75ee6b6efca4a3379f905b8e5c50abea81a747a1752bda18eeba830bd4eR52)